### ISSUE TITLE
Ventura Update

### ISSUE THREAD

**Issue by jakeeverything** [stirtingale/Alfred---Toggle-Function-Keys-On-Of#3]
Love this workflow, but it obviously doesn't work with Ventura. Any chance you can post the update?

**Comment by geraldaddey** [comment]
Also waiting for the Ventura update. Can we contribute to that implementation? I'd love to.

**Comment by stirtingale** [comment]
@geraldaddey

I missed the notification for @jakeeverything update request.

I am not 100% sure how to achieve this as elegantly as before, as Venture have hidden the toggle away now under Keyboard > Keyboard Short Cuts > Function Keys  

![Screenshot 2023-05-15 at 09 10 15](https://github.com/stirtingale/Alfred---Toggle-Function-Keys-On-Of/assets/25981099/53f4ca21-ae3a-47f4-94e6-2421fde52182)

If you want to have a crack at it I am happy to accept the pull-request.

**Comment by geraldaddey** [comment]
Yeah you're right. Ventura hides it. Maybe this can help. Open any pane in settings and run the script to view it's location.

`tell application "System Events"
	tell application process "System Settings"
		set aWindow to window 1
	end tell
end tell
`


<img width="1417" alt="screenshot  2023-05-15 at 9 49 09 AM" src="https://github.com/stirtingale/Alfred---Toggle-Function-Keys-On-Of/assets/42818198/3943793c-b978-4624-b4f3-a46d917201bb">

**Comment by stirtingale** [comment]
@geraldaddey 

Figured it out I think?

You can open with https://developer.apple.com/documentation/devicemanagement/systempreferences

Then it's just a matter of figuring out where to go / toggle

```
do shell script "open x-apple.systempreferences:com.apple.preference.keyboard?FunctionKeys"

tell application "System Events"
	tell application process "System Settings"
		delay 1
		set timeoutSeconds to 2.0
		set uiScript to "click checkbox 1 of group 1 of scroll area 1 of group 2 of splitter group 1 of group 1 of sheet 1 of window \"Keyboard\" of application process \"System Settings\""
		my doWithTimeout(uiScript, timeoutSeconds)
		set uiScript to "click UI Element 2 of group 2 of splitter group 1 of group 1 of sheet 1 of window \"Keyboard\" of application process \"System Settings\""
		my doWithTimeout(uiScript, timeoutSeconds)
	end tell
	delay 0.5
	tell application "System Settings"
		quit
	end tell
end tell


on doWithTimeout(uiScript, timeoutSeconds)
	set endDate to (current date) + timeoutSeconds
	repeat
		try
			run script "tell application \"System Events\"
" & uiScript & "
end tell"
			exit repeat
		on error errorMessage
			if ((current date) > endDate) then
				error "Can not " & uiScript
			end if
		end try
	end repeat
end doWithTimeout


```

Alfred doesn't like doing that in one go and doesn't seem to support the timeout function so I have written as a two step process within afred, a shell to open that page, then apple script to toggle and close. 

That works for me. I have updated the main repo @ toggle-function-keys-ventura.alfredworkflow

**Comment by geraldaddey** [comment]
Great! Thanks a lot.
